src/Sulu/Bundle/SnippetBundle/Controller/SnippetAreaController.php

```diff
@@ -54,11 +55,17 @@ public function cgetAction(Request $request)
                 'key' => $key,
                 'template' => $area['template'],
                 'title' => $area['title'],
+                'defaultUuid' => null,
+                'defaultTitle' => null,
             ];
 
-            $snippet = $defaultSnippetManager->load($webspaceKey, $key, $this->getUser()->getLocale());
-            $areaData['defaultUuid'] = $snippet ? $snippet->getUuid() : null;
-            $areaData['defaultTitle'] = $snippet ? $snippet->getTitle() : null;
+            try {
+                $snippet = $defaultSnippetManager->load($webspaceKey, $key, $this->getUser()->getLocale());
```

Comment:
@wachterjohannes should we not just avoid the load request if webspaceKey is empty?

Reply:
the webspace-key is mandatory - the problem is, that the loaded snippet is from the wrong type. I will add in this PR that there is a wrong snippet selected  so that we can show an error message in the javascript part.

Reply:
Should we not remove it from the default snippet when a template switch was done?

Reply:
i am not sure - because the information that there is something (invalid) selected is gone forever. @danrot @chirimoya what do you think?

Reply:
@wachterjohannes ok thinking about it again we should probably show it in the UI under default snippets that there need to be choose a new snippet or?

Reply:
I am not really sure what the ideal solution here is... Maybe warn when deleting the snippet? I think we already do something similar when deleting the snippet.